src/Geralt/Crypto/Argon2id.cs

```diff
@@ -54,10 +76,29 @@ public static bool NeedsRehash(ReadOnlySpan<byte> hash, int iterations, int memo
         return ret == -1 ? throw new FormatException("Invalid encoded password hash.") : ret == 1;
     }
 
+    public static bool NeedsRehash(string hash, int iterations, int memorySize)
+    {
+        Validation.NotNull(nameof(hash), hash);
```

Comment:
Please change this to `Validation.NotNullOrEmpty()`.

Reply:
Again, it's probably better to do `Validation.NotNull()` and then `Validation.SizeBetween()` with the `MinHashSize` and `MaxHashSize` constants.